ios/Exponent/Versioned/Modules/Api/EXContacts.m

```diff
@@ -53,48 +53,77 @@ - (void)dealloc
+    contact[@"lastName"] = (__bridge_transfer NSString *)(ABRecordCopyValue(person, kABPersonLastNameProperty));
+    contact[@"middleName"] = (__bridge_transfer NSString *)(ABRecordCopyValue(person, kABPersonMiddleNameProperty));
+    contact[@"name"] = [self _assembleDisplayNameFromFirstName:contact[@"firstName"] lastName:contact[@"lastName"]];
+    contact[@"company"] = (__bridge_transfer NSString *)(ABRecordCopyValue(person, kABPersonOrganizationProperty));
+    contact[@"jobTitle"] = (__bridge_transfer NSString *)(ABRecordCopyValue(person, kABPersonJobTitleProperty));
     
-    if ([fieldsSet containsObject:@"phone_number"]) {
-      ABMultiValueRef phoneNumbers = ABRecordCopyValue(person, kABPersonPhoneProperty);
-      if (phoneNumbers) {
-        if (ABMultiValueGetCount(phoneNumbers) > 0) {
-          contact[@"phoneNumber"] = (__bridge_transfer NSString *)ABMultiValueCopyValueAtIndex(phoneNumbers, 0);
+    ABMultiValueRef phoneNumbers = ABRecordCopyValue(person, kABPersonPhoneProperty);
+    if (phoneNumbers) {
+      CFIndex numberOfPhones = ABMultiValueGetCount(phoneNumbers);
+      
+      if (numberOfPhones > 0) {
+        contact[@"phoneNumbers"] = [NSMutableArray new];
+        
+        for (NSUInteger index = 0; index < numberOfPhones; index++) {
+          NSString *phoneNumber = (__bridge_transfer NSString *)(ABMultiValueCopyValueAtIndex(phoneNumbers, index));
+          
+          CFStringRef phoneLabelRef = ABMultiValueCopyLabelAtIndex(phoneNumbers, index);
+          NSString *phoneLabel = phoneLabelRef ?
+            (__bridge_transfer NSString *)(ABAddressBookCopyLocalizedLabel(phoneLabelRef)) :
+            nil;
+          
+          [contact[@"phoneNumbers"] addObject:@{ @"number": phoneNumber, @"label": phoneLabel }];
+          
+          if (phoneLabelRef) {
+            CFRelease(phoneLabelRef);
+          }
         }
-        CFRelease(phoneNumbers);
       }
+      
+      CFRelease(phoneNumbers);
     }
     
-    if ([fieldsSet containsObject:@"email"]) {
-      ABMultiValueRef emails = ABRecordCopyValue(person, kABPersonEmailProperty);
-      if (emails) {
-        if (ABMultiValueGetCount(emails) > 0) {
-          contact[@"email"] = (__bridge_transfer NSString *)ABMultiValueCopyValueAtIndex(emails, 0);
+    ABMultiValueRef emails = ABRecordCopyValue(person, kABPersonEmailProperty);
+    if (emails) {
+      CFIndex numberOfEmails = ABMultiValueGetCount(emails);
+      
+      if (numberOfEmails > 0) {
+        contact[@"emails"] = [NSMutableArray new];
+        
+        for (NSUInteger index = 0; index < numberOfEmails; index++) {
+          NSString *emailAddress = (__bridge_transfer NSString *)(ABMultiValueCopyValueAtIndex(emails, index));
+          
+          CFStringRef emailLabelRef = ABMultiValueCopyLabelAtIndex(emails, index);
+          NSString *emailLabel = emailLabelRef ?
+            (__bridge_transfer NSString *)(ABAddressBookCopyLocalizedLabel(emailLabelRef)) :
```

Comment:
I've been always indenting on tenary, like here. Xcode keeps aligning that line with `NSString` on 114, is that against our conventions?